R/utils.R

```diff
@@ -0,0 +1,189 @@
+#' Sample names using [randomNames::randomNames()]
+#'
+#' @description
+#' Sample names for specified sexes by sampling with replacement to avoid
+#' exhausting number of name when `sample.with.replacement = FALSE`. The
+#' duplicated names during sampling need to be removed to ensure each
+#' individual has a unique name. In order to have enough unique names, more
+#' names than required are sampled from [randomNames()], and the level of
+#' oversampling is determined by the `buffer_factor` argument. A
+#' `buffer_factor` too high and the more names are sampled which takes longer,
+#' a `buffer_factor` too low and not enough unique names are sampled and
+#' the `.sample_names()` function will need to loop until it has enough
+#' unique names.
+#'
+#' @inheritParams .add_date
+#' @param buffer_factor A single `numeric` determining the level of
+#' oversampling (or buffer) when creating a vector of unique names from
+#' [randomNames()].
+#'
+#' @return A `character` vector.
+#' @keywords internal
+.sample_names <- function(.data,
+                          buffer_factor = 1.5) {
+  m_lgl_idx <- .data$sex == "m"
+  f_lgl_idx <- .data$sex == "f"
+  num_m <- sum(m_lgl_idx)
+  num_f <- sum(f_lgl_idx)
+  num_sample_m <- ceiling(num_m * buffer_factor)
+  num_sample_f <- ceiling(num_f * buffer_factor)
+
+  # create sample of names so there are no duplicates
+  names_m <- character(0)
+  while (length(names_m) < num_m) {
+    names_m <- unique(
+      randomNames::randomNames(
+        which.names = "both",
+        name.sep = " ",
+        name.order = "first.last",
+        gender = rep("M", num_sample_m),
+        sample.with.replacement = TRUE
+      )
+    )
+  }
+
+  names_f <- character(0)
+  while (length(names_f) < num_f) {
+    names_f <- unique(
+      randomNames::randomNames(
+        which.names = "both",
+        name.sep = " ",
+        name.order = "first.last",
+        gender = rep("F", num_sample_f),
+        sample.with.replacement = TRUE
+      )
+    )
+  }
+
+  # subset to use required number of names
+  names_m <- names_m[1:num_m]
+  names_f <- names_f[1:num_f]
+
+  # order names with sex codes from .data
+  names_mf <- vector(mode = "character", length = nrow(.data))
+  names_mf[m_lgl_idx] <- names_m
+  names_mf[f_lgl_idx] <- names_f
+
+  # return vector of names
+  names_mf
+}
+
+#' Anonymise names
```

Comment:
I notice I expect this to be `.anonymize` and I realize that we do not have any conventions for naming within Epiverse. I don't want to start a language discussion as I know this is personal preference, but wonder what your thoughts are on this? Do you expect functions to be named in en-GB, en-US or not do you have no implicit assumptions for yourself?

Reply:
I've chosen to use en-GB as the language for the package: https://github.com/epiverse-trace/simulist/blob/main/DESCRIPTION#L50. This impacts the spell checking we perform in the package testing. 

There is no Epiverse policy on which language to pick (see https://github.com/epiverse-trace/packagetemplate/issues/103). There was also a discussion on the internal slack which I can send you if you're interested. 

My personal policy is that if the function is exported and there is a difference in spelling between US and GB spelling then I export both. See https://github.com/epiverse-trace/epiparameter/blob/main/R/epidist.R#L674-L680. (Tidyverse do the same). For internal functions like `.anonymise()` I only have a GB spelling, but would not have a problem if other developers wanted their internal functions to use en-US.